### PR TITLE
check flavor and reject CPUs less than 2

### LINKED ISSUE #1450: Handle scenario when flavour is too small
/kind bug

**What steps did you take and what happened:**
When trying to launch a cluster with little number of vCPUs, CAPO does not validate it and therefore the cluster won't build because of `kubeadm` failing with this:

```
[2023-01-17 18:01:50] [preflight] Running pre-flight checks
[2023-01-17 18:01:51] error execution phase preflight: [preflight] Some fatal errors occurred:
[2023-01-17 18:01:51] 	[ERROR NumCPU]: the number of available CPUs 1 is less than the required 2
```

**What did you expect to happen:**
Since this is an expected failure case and the only way to find out that this was broken is by SSH-ing into the system and looking at cloud-init logs, it would be nice for CAPO to simply refuse to spawn the system if the flavour has 1 CPU only.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster API Provider OpenStack version (Or `git rev-parse HEAD` if manually built):
- Cluster-API version:
- OpenStack version:
- Minikube/KIND version:
- Kubernetes version (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

check flavor and reject flavor with VCPUs < 2 to align with kubeadm checks 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1450 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: pkg/cloud/services/compute/instance.go [hunk 1]
```diff
@@ -126,11 +127,23 @@ func (s *Service) constructNetworks(openStackCluster *infrav1.OpenStackCluster,
...
 }
 
-func (s *Service) createInstanceImpl(eventObject runtime.Object, openStackCluster *infrav1.OpenStackCluster, instanceSpec *InstanceSpec, clusterName string, retryInterval time.Duration) (*InstanceStatus, error) {
+func (s *Service) getAndValidateFlavor(flavorName string, isBastion bool) (*flavors.Flavor, error) {
+	f, err := s.getComputeClient().GetFlavorFromName(flavorName)
+	if err != nil {
+		return nil, fmt.Errorf("error getting flavor from flavor name %s: %v", flavorName, err)
+	}
+	if !isBastion && f.VCPUs <= 1 {
+		return nil, fmt.Errorf("error create machines with flavor's vCPU less than 2, this will cause kubeadm fail to start, update your flavor setting to make VCPUS equal to or bigger than 2")
```
Comment (lentzi90) [inline pkg/cloud/services/compute/instance.go:140]: I found this a bit hard to understand, would suggest something like this instead:
```suggestion
		return nil, fmt.Errorf("kubeadm requires a minimum of 2 vCPUs, pick a flavor with at least 2 vCPUs")
```

Comment (jichenjc) [reply pkg/cloud/services/compute/instance.go:140]: ok, will update

**Review by lentzi90 — COMMENTED** [review]
/lgtm

**Review by mnaser — APPROVED** [review]

**Pull request merged**